src/datepicker/datepicker.js

```diff
@@ -616,6 +615,11 @@ function ($compile, $parse, $document, $position, dateFilter, dateParser, datepi
 
       function validator(modelValue, viewValue) {
         var value = modelValue || viewValue;
+
+        if (!attrs.ngRequired && !value) {
```

Comment:
I wasn't saying just working - I was wondering whether the validator even runs if `!attrs.ngRequired`. If it doesn't run, it wouldn't be necessary, but if it does, then this line is fine.